haystack/nodes/label_generator/pseudo_label_generator.py

```diff
@@ -0,0 +1,211 @@
+    |       train_examples.append(item)
+    |
+    |   # Users can also provide a list of questions/document dictionary pairs to generate pseudo labels for
+    |
+    |   question_doc_pairs: List[Dict[str, str]] = [{"question": "question text ...", "document": "document text ..."}]
+    |   plg = PseudoLabelGenerator(question_doc_pairs, retriever)
+    |   train_examples = []
+    |   for idx, doc in enumerate(document_store):
+    |   output_samples = plg.run(documents=[doc])
+    |   for item in output_samples:
+    |       train_examples.append(item)
+    ```
+    """
+
+    def __init__(
+        self,
+        question_producer: Union[QuestionGenerator, List[Dict[str, str]]],
+        retriever: BaseRetriever,
+        cross_encoder_model_name_or_path: str = "cross-encoder/ms-marco-MiniLM-L-6-v2",
+        max_questions_per_document: int = 3,
+        top_k: int = 50,
+    ):
+        """
+        Loads the cross encoder model and prepares PseudoLabelGenerator.
+
+        :param question_producer: The question producer used to generate questions or a list of already produced
+        questions/document pairs in Dict format {"question": "question text ...", "document": "document text ..."}.
+        :type question_producer: Union[QuestionGenerator, List[Dict[str, str]]]
+        :param retriever: The retriever used to query document stores
+        :type retriever: BaseRetriever
+        :param cross_encoder_model_name_or_path: The path to the cross encoder model, defaults to
+        cross-encoder/ms-marco-MiniLM-L-6-v2
+        :type cross_encoder_model_name_or_path: str (optional)
+        :param max_questions_per_document: The max number of questions generated per document, defaults to 3
+        :type max_questions_per_document: int
+        :param top_k: The number of answers retrieved for each question, defaults to 50
+        :type top_k: int (optional)
+        """
+
+        super().__init__()
+        self.question_document_pairs = None
+        self.question_generator = None
+        if isinstance(question_producer, QuestionGenerator):
+            self.question_generator = question_producer
+        elif isinstance(question_producer, list) and len(question_producer) > 0:
+            example = question_producer[0]
+            if isinstance(example, dict) and "question" in example and "document" in example:
+                self.question_document_pairs = question_producer
+            else:
+                raise ValueError("question_producer list must contain dicts with keys 'question' and 'document'")
+        else:
+            raise ValueError("Provide either a QuestionGenerator or nonempty list of questions/document pairs")
+
+        self.retriever = retriever
+        self.cross_encoder = CrossEncoder(cross_encoder_model_name_or_path)
+        self.max_questions_per_document = max_questions_per_document
+        self.top_k = top_k
+
+    def generate_questions(self, documents: List[Document]) -> List[Dict[str, str]]:
```

Comment:
Maybe we can add an optional `batch_size` parameter here that is passed to `generate_batch`. If no `batch_size` is set here, we can take the `batch_size` set in the init method.